services/web/server/src/simcore_service_webserver/resource_manager/garbage_collector.py

```diff
@@ -73,9 +81,53 @@ async def collect_garbage(registry: RedisResourceRegistry, app: web.Application)
                 )
 
                 await remove_resources_if_guest_user(
-                    app=app, project_uuid=resource_value, user_id=int(key["user_id"])
+                    app=app,
+                    project_uuid=resource_value,
+                    user_id=int(dead_key["user_id"]),
                 )
 
+    # remove possible pending contianers
+    await remove_orphaned_services(registry, app)
+
+
+async def remove_orphaned_services(
+    registry: RedisResourceRegistry, app: web.Application
+) -> None:
+    """Removes services which are no longer tracked in the database
+
+    Multiple deployments can be active at the same tim on the same cluster.
+    This will also check the current SWARM_STACK_NAME of the service which
+    must be matching its own.
+
+    If the service is a dynamic service
+    """
+    logger.info("Starting orphaned services removal...")
+    currently_opened_projects = set()
+    alive_keys, _ = await registry.get_all_resource_keys()
+    for alive_key in alive_keys:
+        resources = await registry.get_resources(alive_key)
+        if "project_id" not in resources:
+            continue
+
+        node_id = await get_node_id_from_project_id(app, resources["project_id"])
```

Comment:
Yes, and it make no difference when searching for a uuid which always has a value, it will not match. I've also picked up this one again today. I should have put a comment.